### DISCUSSION TITLE
SDK-Python: Get all prompts from LangfuseSDK Client

### DISCUSSION THREAD

**Discussion by nthanapaisal** [langfuse/langfuse#4009] [Ideas]
### Describe the feature or potential improvement

hi, 

I wonder if there is a possibility of getting a feature to retrieve all prompts from Langfuse Client instead of using the API. We can only get 1 prompt using langfuse_client.get_prompt(name="str"). Or am I missing something? 

Thank you

**Comment by nthanapaisal** [comment]
Sorry All good i found the way

1 .create langfuse client using SDK
2. use client.prompts.list()

langfuse = Langfuse(...)
prompts = langfuse.client.prompts.list()

thanks!

**Reply by nthanapaisal** [reply]
ok but it doesn't return prompt contents just prompts info? the way around would be compile all names from that prompts list above and call get_prompt one at the time...?

**Reply by marcklingen** [reply]
that's correct, you'll need to get the individual prompt via the api/sdk.

The list endpoint does not include prompt values as prompts can be really large which would make the performance of this endpoint less predictable.

**Reply by Youddha** [reply]
The problem is it only return the  first prompt. Can client.prompts.list() access page 2 ? or can we increase the limit there ?

edit: It's nowhere in the documentation but  client.prompts.list() does accept limit=x where x is max 100. example `client.prompts.list(limit=100)` for prompt 101+ you would need to implement pagination.

**Reply by marcklingen** [reply]
That's correct, the sdk exposes a typed wrapper of the api, please see api reference for more details:

- https://langfuse.com/docs/api#access-via-sdks
- https://api.reference.langfuse.com
